neurolib/models/multimodel/builder/adex.py

```diff
@@ -0,0 +1,944 @@
+        :type connectivity: np.ndarray
+        :param delays: local delay matrix
+        :type delays: np.ndarray
+        :param exc_seed: seed for random number generator for the excitatory
+            mass
+        :type exc_seed: int|None
+        :param inh_seed: seed for random number generator for the inhibitory
+            mass
+        :type inh_seed: int|None
+        """
+        excitatory_mass = ExcitatoryAdExMass(
+            params=exc_params, lin_nonlin_cascade_filename=exc_lin_nonlin_cascade_filename, seed=exc_seed
+        )
+        excitatory_mass.index = 0
+        inhibitory_mass = InhibitoryAdExMass(
+            params=inh_params, lin_nonlin_cascade_filename=inh_lin_nonlin_cascade_filename, seed=inh_seed
+        )
+        inhibitory_mass.index = 1
+        super().__init__(
+            neural_masses=[excitatory_mass, inhibitory_mass], local_connectivity=connectivity, local_delays=delays,
+        )
+        self._rescale_connectivity()
+
+    def update_params(self, params_dict):
+        """
+        Rescale connectivity after params update if connectivity was updated.
+        """
+        rescale_flag = "local_connectivity" in params_dict
+        super().update_params(params_dict)
+        if rescale_flag:
+            self._rescale_connectivity()
+
+    def _sync(self):
+        """
+        Apart from basic EXC<->INH connectivity, construct also squared
+        variants.
+        """
+        connectivity_sq = self.connectivity ** 2 * self.inputs
+        sq_connectivity = [
+            (
+                # exc -> exc squared connectivity
+                self.sync_symbols[f"node_exc_exc_sq_{self.index}"],
+                sum([connectivity_sq[row, col] for row in self.excitatory_masses for col in self.excitatory_masses]),
+            ),
+            (
+                # exc -> inh squared connectivity
+                self.sync_symbols[f"node_inh_exc_sq_{self.index}"],
+                sum([connectivity_sq[row, col] for row in self.inhibitory_masses for col in self.excitatory_masses]),
+            ),
+            (
+                # inh -> exc squared connectivity
+                self.sync_symbols[f"node_exc_inh_sq_{self.index}"],
+                sum([connectivity_sq[row, col] for row in self.excitatory_masses for col in self.inhibitory_masses]),
+            ),
+            (
+                # inh -> inh squared connectivity
+                self.sync_symbols[f"node_inh_inh_sq_{self.index}"],
+                sum([connectivity_sq[row, col] for row in self.inhibitory_masses for col in self.inhibitory_masses]),
+            ),
+        ]
+        return super()._sync() + sq_connectivity
+
+
+class AdExNetwork(Network):
+    """
+    Whole brain network of adaptive exponential integrate-and-fire mean-field
+    excitatory and inhibitory nodes.
+    """
+
+    name = "AdEx mean-field network"
+    label = "AdExNet"
+
+    sync_variables = ["network_exc_exc", "network_exc_exc_sq"]
```

Comment:
just exc->exc in the network (not as in Wilson-Cowan or Wong-Wang), but also a squared variant...